test-crawler/extractor/extractor.go

```diff
@@ -261,6 +260,14 @@ func getFunctionNodes(content string, treeCursor *sitter.TreeCursor, parser *a.P
 				child.Child(4): parameter_list:  (<-chan *types2.RequestEvent, error)
 			*/
 			if child.Type() == string(METHOD_DECLARATION) {
+				comment := content[prevNode.StartByte():prevNode.EndByte()]
```

Comment:
What if the `prevNode` is not of type `COMMENT`? What is its value then?

Reply:
surround with additional check

```
if prevNode.Type() == string(COMMENT) {
  ... then do the logic of parsing and getting content...
    leave it empty string otherwise, so it can be generated based on input and output params
}
```